compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -738,6 +738,12 @@ private BType getMatchingBoundType(BType expType, SymbolEnv env, HashSet<BType>
             return symTable.noType;
         }
 
+        for (SymbolEnv.TypeParamEntry typeParamEntry : env.typeParamsEntries) {
```

Comment:
This seems not the correct solution. In the previous if condition, we check for direct type param matching. If not we are going to resolve structured types. This logic may break this flow.